classes/Validate.php

```diff
@@ -500,7 +500,7 @@ public static function isPasswd($passwd, $size = Validate::PASSWORD_LENGTH)
     public static function isPlaintextPassword($plaintextPasswd, $size = Validate::PASSWORD_LENGTH)
     {
         // The password lenght is limited by `password_hash()`
-        return Tools::strlen($plaintextPasswd) >= $size && Tools::strlen($plaintextPasswd) <= 72;
+        return preg_match('/^(?=.*[A-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^\w])\S{' . $size . ',72}$/', $plaintextPasswd);
```

Comment:
This regex enforces the policy I suppose? Could you please add a comment like "Ensure the password has at least one of x, one of y and one of z"?

Also, I'm wondering why did you use A-z? If you look at an ASCII table, you'll notice that A-z includes the following characters as well:

	[ \ ] ^ _ `

Also, any non A-z character like `é` or `ç` will count as a non word character.

Reply:
For `A-z` it's a mistake I wanted to write `a-z`.
For `é` and `ç`, I consider them as special characters with `(?=.*[^a-zA-Z0-9])`